processor/k8sprocessor/options.go

```diff
@@ -136,7 +136,15 @@ func extractFieldRules(fieldType string, fields ...FieldExtractConfig) ([]kube.F
 	for _, a := range fields {
 		name := a.TagName
 		if name == "" {
-			name = fmt.Sprintf("k8s.pod.%s.%s", fieldType, a.Key)
+			if a.From == kube.MetadataFromPod {
+				name = fmt.Sprintf("k8s.pod.%s.%s", fieldType, a.Key)
+			} else if a.From == kube.MetadataFromNamespace {
+				name = fmt.Sprintf("k8s.namespace.%s.%s", fieldType, a.Key)
+				// By default if the From field is not set for labels and annotations we want to extract them from pod
+			} else if a.From == "" {
+				name = fmt.Sprintf("k8s.pod.%s.%s", fieldType, a.Key)
+				a.From = kube.MetadataFromPod
+			}
```

Comment:
1. We return a helpful error if an invalid value is specified for `From` field like this one: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/3384/files#diff-224a054603fb24c9548928fedba75a1600a41b427a1d12df3b88a847e486ce71R210
2. I think the default needs to be set and error needs to be returned outside this check (if name == "").
3. This might read cleaner as a switch case but that's totally up to you.

Reply:
```go
		if name == "" {
			switch a.From {
			case metdataNamespace:
				name = fmt.Sprintf("k8s.namespace.%s.%s", fieldType, a.Key)
				// By default if the From field is not set for labels and annotations we want to extract them from pod
			case "", metadataPodName:
				name = fmt.Sprintf("k8s.pod.%s.%s", fieldType, a.Key)
				a.From = kube.MetadataFromPod
			default:
				return rules, fmt.Errorf("from must contain exactly one value like pod or namespace")
			}
		}
```

Reply:
This is fine except `a.From = kube.MetadataFromPod` should be before the `if` condition because we don't know where else `a.From` might be used so we shouldn't limit setting the defaults only when `name == ""`. Plus, in general we shouldn't mix figuring out values for two different fields in the same logical block IMO for clarity.